File: push/src/main/java/com/xiaomi/push/mpcd/job/AppIsInstalledCollectionJob.java
package com.xiaomi.push.mpcd.job;

import android.content.Context;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;
import android.text.TextUtils;
import com.xiaomi.channel.commonutils.android.AppInfoUtils;
import com.xiaomi.channel.commonutils.string.Base64Coder;
import com.xiaomi.push.mpcd.Constants;
import com.xiaomi.xmpush.thrift.ClientCollectionType;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/push/mpcd/job/AppIsInstalledCollectionJob.class */
public class AppIsInstalledCollectionJob extends CollectionJob {
    private String mApps;

    public AppIsInstalledCollectionJob(Context context, int i, String str) {
        super(context, i);
        this.mApps = str;
    }

    private String[] revertAppList() {
        if (TextUtils.isEmpty(this.mApps)) {
            return null;
        }
        String strDecodeString = Base64Coder.decodeString(this.mApps);
        if (TextUtils.isEmpty(strDecodeString)) {
            return null;
        }
        return strDecodeString.contains(",") ? strDecodeString.split(",") : new String[]{strDecodeString};
    }

    private static String getInstallerPackageNameCompat(PackageManager packageManager, String str) {
        try {
            Object installSourceInfo = PackageManager.class.getMethod("getInstallSourceInfo", String.class).invoke(packageManager, str);
            if (installSourceInfo == null) {
                return null;
            }
            Object invoke = installSourceInfo.getClass().getMethod("getInstallingPackageName").invoke(installSourceInfo);
            return invoke instanceof String ? (String) invoke : null;
        } catch (ReflectiveOperationException unused) {
            try {
                Object invoke = PackageManager.class.getMethod("getInstallerPackageName", String.class).invoke(packageManager, str);
                return invoke instanceof String ? (String) invoke : null;
            } catch (ReflectiveOperationException e) {
                return null;
            }
        }
    }

    @Override // com.xiaomi.push.mpcd.job.CollectionJob
    public String collectInfo() {
        String[] strArrRevertAppList = revertAppList();
        if (strArrRevertAppList == null || strArrRevertAppList.length <= 0) {
            return null;
        }
        PackageManager packageManager = this.context.getPackageManager();
        StringBuilder sb = new StringBuilder();
        for (String str : strArrRevertAppList) {
            try {
                PackageInfo packageInfo = packageManager.getPackageInfo(str, 16384);
                if (packageInfo != null) {
                    if (sb.length() > 0) {
                        sb.append(Constants.ITEM_SEPARATOR);
                    }
                    String installerPackageName = getInstallerPackageNameCompat(packageManager, str);
                    if (TextUtils.isEmpty(installerPackageName)) {
                        installerPackageName = "null";
                    }
                    sb.append(packageInfo.applicationInfo.loadLabel(packageManager).toString());
                    sb.append(",");
                    sb.append(packageInfo.packageName);
                    sb.append(",");
                    sb.append(packageInfo.versionName);
                    sb.append(",");
                    sb.append(AppInfoUtils.getVersionCode(this.context, packageInfo.packageName));
                    sb.append(",");
                    sb.append(packageInfo.firstInstallTime);
                    sb.append(",");
                    sb.append(packageInfo.lastUpdateTime);
                    sb.append(",");
                    sb.append(installerPackageName);
                }
            } catch (Exception e2) {
            }
        }
        return sb.length() > 0 ? sb.toString() : null;
    }

    @Override // com.xiaomi.push.mpcd.job.CollectionJob
    public ClientCollectionType getCollectionType() {
        return ClientCollectionType.AppIsInstalled;
    }

    @Override // com.xiaomi.channel.commonutils.misc.ScheduledJobManager.Job
    public String getJobId() {
        return "24";
    }
}

File: legacy-runtime/src/main/java/com/xiaomi/push/mpcd/job/AppIsInstalledCollectionJob.kt
package com.xiaomi.push.mpcd.job

import android.content.Context
import android.content.pm.PackageManager
import android.text.TextUtils
import com.xiaomi.channel.commonutils.android.AppInfoUtils
import com.xiaomi.channel.commonutils.string.Base64Coder
import com.xiaomi.push.mpcd.Constants
import com.xiaomi.xmpush.thrift.ClientCollectionType

class AppIsInstalledCollectionJob(
    context: Context,
    period: Int,
    private val mApps: String
) : CollectionJob(context, period) {

    private fun revertAppList(): Array<String>? {
        if (TextUtils.isEmpty(mApps)) return null
        val decoded = Base64Coder.decodeString(mApps)
        if (TextUtils.isEmpty(decoded)) return null
        return if (decoded.contains(",")) decoded.split(",").toTypedArray() else arrayOf(decoded)
    }

    private fun getInstallerPackageNameCompat(pm: PackageManager, pkg: String): String? {
        return try {
            val installSourceInfo = PackageManager::class.java
                .getMethod("getInstallSourceInfo", String::class.java)
                .invoke(pm, pkg) ?: return null
            val result = installSourceInfo::class.java.getMethod("getInstallingPackageName").invoke(installSourceInfo)
            result as? String
        } catch (e: ReflectiveOperationException) {
            try {
                val result = PackageManager::class.java
                    .getMethod("getInstallerPackageName", String::class.java)
                    .invoke(pm, pkg)
                result as? String
            } catch (e: ReflectiveOperationException) {
                null
            }
        }
    }

    override fun collectInfo(): String? {
        val appList = revertAppList() ?: return null
        if (appList.isEmpty()) return null
        val pm = context.packageManager
        val sb = StringBuilder()
        for (pkg in appList) {
            try {
                val packageInfo = pm.getPackageInfo(pkg, 16384) ?: continue
                val appInfo = packageInfo.applicationInfo ?: continue
                if (sb.isNotEmpty()) sb.append(Constants.ITEM_SEPARATOR)
                var installer = getInstallerPackageNameCompat(pm, pkg)
                if (TextUtils.isEmpty(installer)) installer = "null"
                sb.append(appInfo.loadLabel(pm))
                sb.append(",")
                sb.append(packageInfo.packageName)
                sb.append(",")
                sb.append(packageInfo.versionName)
                sb.append(",")
                sb.append(AppInfoUtils.getVersionCode(context, packageInfo.packageName))
                sb.append(",")
                sb.append(packageInfo.firstInstallTime)
                sb.append(",")
                sb.append(packageInfo.lastUpdateTime)
                sb.append(",")
                sb.append(installer)
            } catch (e: Exception) {
            }
        }
        return if (sb.isNotEmpty()) sb.toString() else null
    }

    override fun getCollectionType(): ClientCollectionType = ClientCollectionType.AppIsInstalled

    override fun getJobId(): String = "24"
}
